commons/commons-api/src/test/java/fr/gouv/vitamui/commons/api/utils/MetadataSearchCriteriaUtilsTest.java

```diff
@@ -65,10 +67,35 @@ public void handleSimpleFieldCriteria() throws InvalidCreateOperationException {
             .setValues(List.of(new CriteriaValue().setValue("2023-03-05T23:00:00.000Z")))
             .setDataType(ArchiveSearchConsts.CriteriaDataType.DATE.name());
 
-        MetadataSearchCriteriaUtils.handleSimpleFieldCriteria(queryToFill, searchCriteria);
+        MetadataSearchCriteriaUtils.fillQueryFromCriteriaList(queryToFill, Collections.singletonList(searchCriteria));
+
+        Assertions.assertEquals(
+            "{\"$or\":[{\"$and\":[{\"$or\":[{\"$gt\":{\"ActivationDate\":\"2023-03-06\"}}]}]}]}",
+            queryToFill.toString()
+        );
+    }
+
+    @Test
+    public void handleSimpleDateFieldCriteria() throws InvalidCreateOperationException {
+        BooleanQuery queryToFill = new BooleanQuery(BuilderToken.QUERY.AND);
+        SearchCriteriaEltDto startSearchCriteria = new SearchCriteriaEltDto()
+            .setCriteria("START_DATE")
+            .setCategory(ArchiveSearchConsts.CriteriaCategory.FIELDS)
+            .setOperator(ArchiveSearchConsts.CriteriaOperators.GTE.name())
+            .setValues(List.of(new CriteriaValue().setValue("2023-03-05T23:00:00.000Z")))
+            .setDataType(ArchiveSearchConsts.CriteriaDataType.DATE.name());
+        SearchCriteriaEltDto endSearchCriteria = new SearchCriteriaEltDto()
+            .setCriteria("END_DATE")
+            .setCategory(ArchiveSearchConsts.CriteriaCategory.FIELDS)
+            .setOperator(ArchiveSearchConsts.CriteriaOperators.GTE.name())
+            .setValues(List.of(new CriteriaValue().setValue("2023-03-15T23:00:00.000Z")))
+            .setDataType(ArchiveSearchConsts.CriteriaDataType.DATE.name());
+
+        MetadataSearchCriteriaUtils.fillQueryFromCriteriaList(queryToFill,
+            Arrays.asList(startSearchCriteria, endSearchCriteria));
 
         Assertions.assertEquals(
-            "{\"$or\":[{\"$gt\":{\"ActivationDate\":\"2023-03-05T23:00:00.000Z\"}}]}",
+            "{\"$and\":[{\"$or\":[{\"$gte\":{\"StartDate\":\"2023-03-06\"}}]},{\"$or\":[{\"$gte\":{\"EndDate\":\"2023-03-16\"}}]}]}",
```

Comment:
Est-ce normal d'avoir and -> or -> gte ?

Reply:
Même réponse